akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/HttpCookie.scala

```diff
@@ -113,10 +182,67 @@ final case class HttpCookie(
   /** Java API */
   def withHttpOnly(httpOnly: Boolean): headers.HttpCookie = copy(httpOnly = httpOnly)
   /** Java API */
+  def withSameSite(sameSite: jm.headers.SameSite): headers.HttpCookie = new HttpCookie(name, value, expires, maxAge, domain, path, secure, httpOnly, extension, Option(sameSite.asScala()))
+  /** Java API */
+  def withSameSite(sameSite: Optional[jm.headers.SameSite]): headers.HttpCookie = new HttpCookie(name, value, expires, maxAge, domain, path, secure, httpOnly, extension, sameSite.asScala.map(_.asScala()))
+  /** Java API */
   def withExtension(extension: String): headers.HttpCookie = copy(extension = Some(extension))
 }
 
 object HttpCookie {
+
+  def apply(
+    name:      String,
+    value:     String,
+    expires:   Option[DateTime] = None,
+    maxAge:    Option[Long]     = None,
+    domain:    Option[String]   = None,
+    path:      Option[String]   = None,
+    secure:    Boolean          = false,
+    httpOnly:  Boolean          = false,
+    extension: Option[String]   = None,
+    sameSite:  Option[SameSite] = None
+  ) = new HttpCookie(name, value, expires, maxAge, domain, path, secure, httpOnly, extension, sameSite)
+
+  @deprecated("for binary compatibility", since = "10.2.0")
+  def apply(
+    name:      String,
+    value:     String,
+    expires:   Option[DateTime],
+    maxAge:    Option[Long],
+    domain:    Option[String],
+    path:      Option[String],
+    secure:    Boolean,
+    httpOnly:  Boolean,
+    extension: Option[String]
+  ) = new HttpCookie(name, value, expires, maxAge, domain, path, secure, httpOnly, extension, None)
+
+  @deprecated("for binary compatibility", since = "10.2.0")
+  def unapply(cookie: HttpCookie) = Option((
```

Comment:
(makes sense to me to deprecate this one, but let's indeed not make this one private to keep things source-compatible)